File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        let search = target - nums[i];
        if (map.has(search)) {
            return [map.get(search), i]
        }
        
        if (!map.has(nums[i])) {
            map.set(nums[i], i);
        }
    }   
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # O(n^2)
        # for i in range(0, len(nums)):
        #     for j in range(i+1, len(nums)):
        #         if nums[i] + nums[j] == target:
        #             return [i, j]

        # O(n)
        store = {}
        for i in range(0, len(nums)):
            lookfor = target - nums[i]
            if lookfor in store:
                return [i, store[lookfor]]
            store[nums[i]] = i

